### PR TITLE
Add config option to disable HTML repr

### LINKED ISSUE #9830: Synchronize `__repr__()` and `_repr_html_()`
TIL that `Info` has a dedicated HTML representation, which is 🔥. However, I don't like that it doesn't show the same information as the standard text representation, for example:
Screen Shot 2021-10-11 at 15 50 49
Specifically, I really miss the channel names and I don't like that MEG types are always listed, but ideally there should be no differences at all except for a nicer rendering in HTML. WDYT?

### PR DESCRIPTION
**Pull request by cbrnr**
Fixes #9830.

Open questions:

1. ~~I added four lines of code at the beginning of `_repr_html_()` of each class that defines this method. We could outsource this (in a decorator?), but it's probably overkill right now.~~ I refactored this into a decorator, I think this is cleaner even for the few classes we have.
2. I didn't find a way to set a config option only temporarily (for the current session without writing to the config file) – is this possible somehow? `set_config()` doesn't seem to support it, but maybe I missed a context manager or something similar?

### CODE REVIEW

**Review by agramfort — COMMENTED** [review]
could we test this so the new lines are covered? 🙏 @cbrnr

**Comment by cbrnr** [comment]
Sure, I'll add a test. Does anyone have an answer to my second question? Basically, I'm looking for something like `mne.set_log_level()`, which is only valid within the current session (so not permanently).

File: doc/changes/latest.inc [hunk 1]
```diff
@@ -44,6 +44,7 @@ Enhancements
 - Add example of how to obtain time-frequency decomposition using narrow bandpass Hilbert transforms to :ref:`ex-tfr-comparison` (:gh:`11116` by `Alex Rockhill`_)
 - Add ``==`` and ``!=`` comparison between `mne.Projection` objects (:gh:`11147` by `Mathieu Scheltienne`_)
 - Add ``encoding`` parameter to :func:`mne.io.read_raw_edf` and :func:`mne.io.read_raw_bdf` to support custom (non-UTF8) annotation channel encodings (:gh:`11154` by `Clemens Brunner`_)
+- Add config option ``MNE_REPR_HTML`` to disable HTML repr in notebook environments (:gh:`11159` by `Clemens Brunner`_)
```
Comment (agramfort) [inline doc/changes/latest.inc:47]: looking at the doc and thinking where this information could be added I found

https://mne.tools/stable/auto_tutorials/intro/50_configure_mne.html

and this tells me that maybe using mne config mechanism would be the expect thing to do...

what do you think @cbrnr @drammock ?

Comment (cbrnr) [reply doc/changes/latest.inc:47]: I'm not sure I understand. Do you want to describe the new config option in the docs? We haven't done this for the other existing options though.

**Comment by agramfort** [comment]
what other options do you have in mind?

**Comment by cbrnr** [comment]
If we haven't described our config options so far, I think we should describe all of them. But maybe @drammock knows if there's already some documentation.

**Comment by larsoner** [comment]
> Does anyone have an answer to my second question? Basically, I'm looking for something like mne.set_log_level(), which is only valid within the current session (so not permanently).

The current way to get this behavior is just to set `os.environ['...']`. It will be temporary to that session

**Comment by larsoner** [comment]
... and I don't think there is a single place where we describe all options. Instead we try to describe each option around where it's used, e.g., the 3D options should be described in `set_3d_options`. For this new param since it's more "global" in scope like `set_log_level`, we could just add it to the config tutorial for now, and figure out in a separate issue if we want to add *all* options somewhere else later (I'm not yet convinced we should/need to so we should discuss separately).

**Comment by cbrnr** [comment]
> The current way to get this behavior is just to set `os.environ['...']`. It will be temporary to that session

OK, that's what I did in the test.

Re describing config options, I'm fine with descriptions in different places, but maybe an overview of all possible values would be nice. Maybe even built into `mne.get_config()` somehow? Whatever we decide, I'd vote for tackling this in a separate PR, since this is a bit outside the scope of this PR.

**Comment by drammock** [comment]
> I'm fine with descriptions in different places, but maybe an overview of all possible values would be nice. Maybe even built into `mne.get_config()` somehow?

`mne.get_config("")` returns a list of all valid config keys. (no explanation of what they do though, but fodder for a search engine or `git grep`).

**Comment by cbrnr** [comment]
> `mne.get_config("")` returns a list of all valid config keys. (no explanation of what they do though, but fodder for a search engine or `git grep`).

That's probably good enough for now. WDYT @agramfort?

**Comment by drammock** [comment]
> > `mne.get_config("")` returns a list of all valid config keys. (no explanation of what they do though, but fodder for a search engine or `git grep`).

If we wanted a more thorough documentation of what they all do, a nice way to do it would be for `mne.get_config("")` to return a dict instead of a tuple, with the dict values being a short one-sentence description of what each key does and its valid values.

**Comment by cbrnr** [comment]
I could do that? In this PR?

**Comment by drammock** [comment]
> I could do that? In this PR?

go for it. `mne.get_config("")` is mostly there for informational/interactive purposes, it's only used twice in the codebase in ways that should be compatible with `dict` too:

```console
$ git grep "get_config('')"
mne/utils/tests/test_config.py:24:    assert (len(get_config('')) > 10)  # tuple of valid keys
tutorials/intro/50_configure_mne.py:84:assert 'MNEE_USE_CUUDAA' not in mne.get_config('')
```

**Comment by drammock** [comment]
oh, rereading, maybe you weren't volunteering :)  and even if you were, it should probably be a separate PR.

**Comment by cbrnr** [comment]
That's OK, I'll do it in a separate PR. So I guess this one is ready?

File: mne/utils/misc.py [hunk 2]
```diff
@@ -406,3 +406,32 @@ def _resource_path(submodule, filename):
...
+
+    Returns
+    -------
+    wrapper : function
+        The decorated function.
+    """
+    from ..utils import get_config
+
+    def wrapper(*args, **kwargs):
+        if get_config("MNE_REPR_HTML", "True") == "False":
```
Comment (drammock) [inline mne/utils/misc.py:431]: our config values are always strings, so there's a reasonable chance they might not have proper capitalization. Let's be permissive here by including a `.lower()` and testing against `"false"`

**Comment by cbrnr** [comment]
All green – @agramfort please feel free to merge if all of your comments have been addressed!

**Review by agramfort — APPROVED** [review]

**Comment by agramfort** [comment]
thanks @cbrnr 

can you open issues for the next PRs to complete to follow up? 🙏

**Pull request merged**